### PR TITLE
Add Urdu keyboard and dictionary

### PR DESCRIPTION
**Pull request by mirfatif**
The most commonly used phonetic keyboard layout:
https://en.wikipedia.org/wiki/Urdu_keyboard#Phonetic_Keyboard

Dictionary needs improvement.

### CODE REVIEW

File: addons/languages/urdu/apk/build.gradle
Comment (menny) [inline addons/languages/urdu/apk/build.gradle:1]: this should be set to `false`, until we create the app in the play-store

**Review by menny — CHANGES_REQUESTED** [review]

File: addons/languages/urdu/apk/build.gradle
Comment (mirfatif) [reply addons/languages/urdu/apk/build.gradle:1]: Aren't we going to create the app in the Play Store?

Comment (menny) [reply addons/languages/urdu/apk/build.gradle:1]: yes, we will, but only after this gets merged into master.

Comment (mirfatif) [reply addons/languages/urdu/apk/build.gradle]: Changed to `false`.

**Comment by mirfatif** [comment]
@menny if you are not considering to merge this at the moment, may I publish this addon on Play Store?

**Comment by menny** [comment]
I can merge this and publish. But, first the checks need to be green.
Second, it will take me a bit to publish

**Comment by mirfatif** [comment]
If there something I can do to resolve these, please let me know.
```
* What went wrong:
> Configure project :addons:quicktexts:bbcodes:apk
Execution failed for task ':addons:languages:generatePacksMarkDown'.
Could not find '/tmp/anysoftkeyboard.keystore' file. Release APK will not be signed.
> Cannot get property 'notPublishingReason' on extra properties extension as it does not exist
Building :addons:quicktexts:bbcodes:apk 4.0.1 (or 4.0.1), code 1.
```
```
/usr/bin/docker run --name mennyndk_ask1140_de9776 --label 3b3ac6 --workdir /github/workspace --rm -e TERM -e GRADLE_OPTS -e GRADLE_USER_HOME -e TMPDIR -e INPUT_CODECOV_TOKEN -e HOME -e GITHUB_JOB -e GITHUB_REF -e GITHUB_SHA -e GITHUB_REPOSITORY -e GITHUB_REPOSITORY_OWNER -e GITHUB_RUN_ID -e GITHUB_RUN_NUMBER -e GITHUB_ACTOR -e GITHUB_WORKFLOW -e GITHUB_HEAD_REF -e GITHUB_BASE_REF -e GITHUB_EVENT_NAME -e GITHUB_SERVER_URL -e GITHUB_API_URL -e GITHUB_GRAPHQL_URL -e GITHUB_WORKSPACE -e GITHUB_ACTION -e GITHUB_EVENT_PATH -e RUNNER_OS -e RUNNER_TOOL_CACHE -e RUNNER_TEMP -e RUNNER_WORKSPACE -e ACTIONS_RUNTIME_URL -e ACTIONS_RUNTIME_TOKEN -e ACTIONS_CACHE_URL -e GITHUB_ACTIONS=true -e CI=true --entrypoint "/bin/bash" --network github_network_c997d9c7c9b543da8d296e31f9a5cb88 -v "/var/run/docker.sock":"/var/run/docker.sock" -v "/home/runner/work/_temp/_github_home":"/github/home" -v "/home/runner/work/_temp/_github_workflow":"/github/workflow" -v "/home/runner/work/AnySoftKeyboard/AnySoftKeyboard":"/github/workspace" menny/ndk_ask:1.14.0  ".github/actions/codecov/upload_coverage.sh" "fa64b6e7-aa1b-48e3-ba3b-66cf7fe146d0"
--2020-09-10 07:29:31--  https://codecov.io/bash
Resolving codecov.io (codecov.io)... 35.199.43.247
Connecting to codecov.io (codecov.io)|35.199.43.247|:443... failed: Connection timed out.
Retrying.

--2020-09-10 07:31:43--  (try: 2)  https://codecov.io/bash
Connecting to codecov.io (codecov.io)|35.199.43.247|:443... failed: Connection timed out.
Retrying.

--2020-09-10 07:33:55--  (try: 3)  https://codecov.io/bash
Connecting to codecov.io (codecov.io)|35.199.43.247|:443... failed: Connection timed out.
Retrying.

--2020-09-10 07:36:09--  (try: 4)  https://codecov.io/bash
Connecting to codecov.io (codecov.io)|35.199.43.247|:443... failed: Connection timed out.
Retrying.

--2020-09-10 07:38:24--  (try: 5)  https://codecov.io/bash
Connecting to codecov.io (codecov.io)|35.199.43.247|:443... failed: Connection timed out.
Giving up.
```

**Comment by mubashir-rehman** [comment]
any updates?

**Comment by menny** [comment]
I'm willing to merge this, if @mirfatif says it's ready

**Comment by mirfatif** [comment]
> I'm willing to merge this, if @mirfatif says it's ready

I released it as addon on Play Store: https://play.google.com/store/apps/details?id=com.anysoftkeyboard.languagepack.mirfatif.urdu

Yes it's ready if you want to merge.

**Comment by menny** [comment]
hey @mirfatif , I'm taking another look at this now
Thanks for submitting.

File: addons/languages/urdu/apk/src/main/AndroidManifest.xml
Comment (menny) [inline addons/languages/urdu/apk/src/main/AndroidManifest.xml:2]: do you want to change the `package` to `com.anysoftkeyboard.languagepack.mirfatif.urdu` in the `apk` module? I think it will make things less confusing

Comment (menny) [reply addons/languages/urdu/apk/src/main/AndroidManifest.xml:2]: the `pack` module can stay as is, it doesn't matter

**Comment by menny** [comment]
I pushed a few commits to this PR. @mirfatif you can squash them or, just remove my commits and incorporate the changes in your original commit (as my changes are simple merge conflicts)

**Comment by menny** [comment]
closing this in favor of https://github.com/AnySoftKeyboard/AnySoftKeyboard/pull/3211

Thanks

**Comment by mirfatif** [comment]
Thanks for merging.

**Pull request closed without merging**
